packages/kolibri-tools/lib/i18n/utils.py

```diff
@@ -6,22 +6,74 @@
 import os
 import sys
 
-import kolibri_exercise_perseus_plugin
+try:
+    import kolibri_exercise_perseus_plugin
+
+    PERSEUS_LOCALE_PATH = os.path.join(
+        os.path.dirname(kolibri_exercise_perseus_plugin.__file__), "locale"
+    )
+    PERSEUS_SOURCE_PATH = os.path.join(PERSEUS_LOCALE_PATH, "en", "LC_MESSAGES")
+    if not os.path.exists(PERSEUS_SOURCE_PATH):
+        os.makedirs(PERSEUS_SOURCE_PATH)
+    if not os.path.exists(PERSEUS_LOCALE_PATH):
+        os.makedirs(PERSEUS_LOCALE_PATH)
+except:
+    PERSEUS_LOCALE_PATH = None
+    PERSEUS_SOURCE_PATH = None
+    pass
 
 
 logging.basicConfig(format="%(levelname)s: %(message)s", level=logging.INFO)
 logging.StreamHandler(sys.stdout)
 
 
-LOCALE_PATH = os.path.abspath(
-    os.path.join(os.path.dirname(__file__), os.pardir, os.pardir, "kolibri", "locale")
-)
+PROJECT_NAME = os.getenv("CROWDIN_PROJECT", "kolibri")
+
+
+def calculated_locale_path():
+    path_to_here = os.path.dirname(__file__)
+    if "node_modules" in path_to_here:
+        locale_path = os.path.abspath(
+            os.path.join(
+                path_to_here,
+                os.pardir,
+                os.pardir,
+                os.pardir,
+                os.pardir,
+                os.pardir,
+                os.pardir,
+                "{}-locale".format(PROJECT_NAME),
+            )
+        )
+    else:
+        locale_path = os.path.abspath(
+            os.path.join(
+                path_to_here,
+                os.pardir,
+                os.pardir,
+                os.pardir,
+                os.pardir,
+                "{}-locale".format(PROJECT_NAME),
+            )
+        )
+    if not os.path.exists(locale_path):
+        os.makedirs(locale_path)
+
+    return locale_path
```

Comment:
this seems fairly brittle, and I'm not entirely sure what it's supposed to do

Reply:
Yeah, I am not quite sure what this is meant to be doing or why `node_modules` is particularly meaningful here, but using that as a sign post feels like a code smell.

Reply:
Yeah this stinks to high heaven. I'm going to clean this all up a bit and make things dependent on env vars entirely to avoid the kind of confusion I ran into while working on it :-D